blocks/library/gallery/index.js

```diff
@@ -119,21 +143,26 @@ registerBlockType( 'core/gallery', {
 		}
 
 		return (
-			<div className={ `blocks-gallery align${ align }` }>
-				{ images.map( ( img, i ) => (
-					<GalleryImage key={ i } img={ img } />
+			<div className={ `blocks-gallery align${ align } columns-${ columns }` }>
+				{ images.map( ( img ) => (
+					<GalleryImage key={ img.url } img={ img } />
 				) ) }
+				{ focus && images.length > 1 &&
+					<InspectorControls>
+						<label className="blocks-text-control__label" htmlFor={ rangeId }>{ __( 'Columns' ) }</label>
+						<input id={ rangeId } type="range" min="1" max={ Math.min( MAX_COLUMNS, images.length ) } value={ columns } onChange={ setColumnsNumber } />
+						<span>{columns}</span>
```

Comment:
Could be a nice improvement to make the slider control generic (the same way we do for text) https://github.com/WordPress/gutenberg/blob/master/blocks/inspector-controls/text-control/index.js